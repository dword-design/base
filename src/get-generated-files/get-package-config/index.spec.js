import tester from '@dword-design/tester'
import testerPluginTmpDir from '@dword-design/tester-plugin-tmp-dir'
import execa from 'execa'
import fs from 'fs-extra'
import outputFiles from 'output-files'

import { Base } from '@/src/index.js'

export default tester(
  {
    'custom config': async () => {
      await fs.outputFile(
        'package.json',
        JSON.stringify({
          baseConfig: 'foo',
          devDependencies: {
            'base-config-foo': '^1.0.0',
          },
        })
      )

      const base = new Base({ packageConfig: { main: 'dist/index.scss' } })
      expect(base.getPackageConfig().main).toEqual('dist/index.scss')
    },
    deploy: async () => {
      await fs.outputFile('package.json', JSON.stringify({ deploy: true }))

      const base = new Base()

      const packageConfig = base.getPackageConfig()
      expect(packageConfig.deploy).toBeTruthy()
    },
    empty() {
      expect(new Base().getPackageConfig()).toMatchSnapshot(this)
    },
    async 'existing package'() {
      await fs.outputFile(
        'package.json',
        JSON.stringify({
          author: 'foo bar',
          bin: {
            foo: './dist/cli.js',
          },
          dependencies: {
            foo: '^1.0.0',
          },
          description: 'foo bar',
          devDependencies: {
            'base-config-bar': '^1.0.0',
          },
          extra: 'foo',
          files: 'foo',
          keywords: ['foo', 'bar'],
          license: 'ISC',
          main: 'dist/index.scss',
          name: 'foo',
          peerDependencies: {
            nuxt: '^1.0.0',
          },
          publishConfig: {
            access: 'public',
          },
          scripts: {
            foo: 'echo \\"foo\\"',
            test: 'echo \\"foo\\"',
          },
          types: 'types.d.ts',
          version: '1.1.0',
        })
      )
      expect(new Base().getPackageConfig()).toMatchSnapshot(this)
    },
    async 'git repo'() {
      await execa.command('git init')
      await execa.command('git remote add origin git@github.com:bar/foo.git')
      expect(new Base().getPackageConfig()).toMatchSnapshot(this)
    },
    'non-github repo': async () => {
      await execa.command('git init')
      await execa.command('git remote add origin git@special.com:bar/foo.git')
      expect(() => new Base().getPackageConfig()).toThrow(
        'Only GitHub repositories are supported.'
      )
    },
    private: () =>
      expect(new Base().getPackageConfig({ private: true })).toBeTruthy(),
    async 'sub-folder'() {
      await execa.command('git init')
      await execa.command('git remote add origin git@github.com:bar/foo.git')
      await outputFiles({
        test: {},
      })
      process.chdir('test')
      expect(new Base().getPackageConfig()).toMatchSnapshot(this)
    },
    'types.d.ts': async () => {
      await fs.outputFile('types.d.ts', '')

      const packageConfig = new Base().getPackageConfig()
      expect(packageConfig.files).toEqual(['dist', 'types.d.ts'])
    },
  },
  [testerPluginTmpDir()]
)
