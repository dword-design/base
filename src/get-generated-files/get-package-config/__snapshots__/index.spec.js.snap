// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index empty 1`] = `
Object {
  "author": "Sebastian Landwehr <info@sebastianlandwehr.com>",
  "engines": Object {
    "node": ">=14",
  },
  "files": Array [
    "dist",
  ],
  "funding": "https://github.com/sponsors/dword-design",
  "license": "MIT",
  "main": "dist/index.js",
  "publishConfig": Object {
    "access": "public",
  },
  "scripts": Object {
    "checkUnknownFiles": "base checkUnknownFiles",
    "commit": "base commit",
    "depcheck": "base depcheck",
    "dev": "base dev",
    "lint": "base lint",
    "prepare": "base prepare",
    "prepublishOnly": "base prepublishOnly",
    "test": "base test",
  },
  "version": "1.0.0",
}
`;

exports[`index existing package 1`] = `
Object {
  "author": "Sebastian Landwehr <info@sebastianlandwehr.com>",
  "bin": Object {
    "foo": "./dist/cli.js",
  },
  "dependencies": Object {
    "foo": "^1.0.0",
  },
  "description": "foo bar",
  "devDependencies": Object {
    "base-config-bar": "^1.0.0",
  },
  "engines": Object {
    "node": ">=14",
  },
  "files": Array [
    "dist",
  ],
  "funding": "https://github.com/sponsors/dword-design",
  "keywords": Array [
    "foo",
    "bar",
  ],
  "license": "MIT",
  "main": "dist/index.js",
  "name": "foo",
  "peerDependencies": Object {
    "nuxt": "^1.0.0",
  },
  "publishConfig": Object {
    "access": "public",
  },
  "scripts": Object {
    "checkUnknownFiles": "base checkUnknownFiles",
    "commit": "base commit",
    "depcheck": "base depcheck",
    "dev": "base dev",
    "lint": "base lint",
    "prepare": "base prepare",
    "prepublishOnly": "base prepublishOnly",
    "test": "base test",
  },
  "types": "types.d.ts",
  "version": "1.1.0",
}
`;

exports[`index git repo 1`] = `
Object {
  "author": "Sebastian Landwehr <info@sebastianlandwehr.com>",
  "engines": Object {
    "node": ">=14",
  },
  "files": Array [
    "dist",
  ],
  "funding": "https://github.com/sponsors/dword-design",
  "license": "MIT",
  "main": "dist/index.js",
  "publishConfig": Object {
    "access": "public",
  },
  "repository": "dword-design/foo",
  "scripts": Object {
    "checkUnknownFiles": "base checkUnknownFiles",
    "commit": "base commit",
    "depcheck": "base depcheck",
    "dev": "base dev",
    "lint": "base lint",
    "prepare": "base prepare",
    "prepublishOnly": "base prepublishOnly",
    "test": "base test",
  },
  "version": "1.0.0",
}
`;

exports[`index sub-folder 1`] = `
Object {
  "author": "Sebastian Landwehr <info@sebastianlandwehr.com>",
  "engines": Object {
    "node": ">=14",
  },
  "files": Array [
    "dist",
  ],
  "funding": "https://github.com/sponsors/dword-design",
  "license": "MIT",
  "main": "dist/index.js",
  "publishConfig": Object {
    "access": "public",
  },
  "scripts": Object {
    "checkUnknownFiles": "base checkUnknownFiles",
    "commit": "base commit",
    "depcheck": "base depcheck",
    "dev": "base dev",
    "lint": "base lint",
    "prepare": "base prepare",
    "prepublishOnly": "base prepublishOnly",
    "test": "base test",
  },
  "version": "1.0.0",
}
`;
